driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java

```diff
@@ -140,7 +140,7 @@ protected ClusterableServer getServer(final ServerAddress serverAddress) {
     private final class DefaultServerStateListener extends ServerListenerAdapter {
         @Override
         public void serverDescriptionChanged(final ServerDescriptionChangedEvent event) {
-            onChange(event);
+            onChange(event, event.getShouldEventBePublished());
```

Comment:
Nit: Do you need to change the signature? You already pass in the event.

Reply:
Done